Enforce slug constraints in API

dcramer opened the pull request:
Most slug serialization endpoints were using a simple CharField, which was allowing invalid slugs. This switches them to RegexField with the restrictive charset.

As a follow up it would be useful to abstract things into a SlugField.

@getsentry/infrastructure

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/getsentry/sentry/3924)
<!-- Reviewable:end -->

mattrobenolt commented:
Do we not allow capital letters? The url routes use `[\w_-]+` which is the same as `[a-zA-Z0-9_-]+`. So this regex would be a bit more restrictive.

---

Review status: 0 of 6 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3924#-:-KOw1E4eWyE1DMyH54bu:blqwd56)*
<!-- Sent from Reviewable.io -->

dcramer commented:
I think we should not allow capital letters going forward, but I'm open to discussion.

mattrobenolt commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/sentry/api/endpoints/organization_details.py, line 50 \[r1\]](https://reviewable.io:443/reviews/getsentry/sentry/3924#-KOw1fT753r5w-l1F3yy:-KOw1fT753r5w-l1F3yz:b7hv4mk) ([raw file](https://github.com/getsentry/sentry/blob/3edefbd6fc69e60ee8634a804c504d66fcd12b11/src/sentry/api/endpoints/organization_details.py#L50)):*
> ```Python
> class OrganizationSerializer(serializers.ModelSerializer):
>     projectRateLimit = serializers.IntegerField(min_value=1, max_value=100)
>     slug = serializers.RegexField(r'^[a-z0-9_\-]+$', max_length=64,
> ```

The actual database field, `SlugField` in Django, is only a length of 50. So these should all reflect that.

---

*[src/sentry/api/endpoints/organization_index.py, line 32 \[r1\]](https://reviewable.io:443/reviews/getsentry/sentry/3924#-KOw1p0V0C8rgRzMxdeG:-KOw1p0V0C8rgRzMxdeH:bh0y53y) ([raw file](https://github.com/getsentry/sentry/blob/3edefbd6fc69e60ee8634a804c504d66fcd12b11/src/sentry/api/endpoints/organization_index.py#L32)):*
> ```Python
> 
> class OrganizationSerializer(serializers.Serializer):
>     name = serializers.CharField(max_length=200, required=True)
> ```

I know this preexisted, but this field length is only 64 in the databass.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3924)*
<!-- Sent from Reviewable.io -->

dcramer commented:
Review status: 0 of 6 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/sentry/api/endpoints/organization_details.py, line 50 \[r1\]](https://reviewable.io:443/reviews/getsentry/sentry/3924#-KOw1fT753r5w-l1F3yy:-KOwChSGq-nt_bZhc1LC:b-7ql1td) ([raw file](https://github.com/getsentry/sentry/blob/3edefbd6fc69e60ee8634a804c504d66fcd12b11/src/sentry/api/endpoints/organization_details.py#L50)):*
<details><summary><i>Previously, mattrobenolt (Matt Robenolt) wrote…</i></summary>
> The actual database field, `SlugField` in Django, is only a length of 50. So these should all reflect that.
</details>
fixed

---

*[src/sentry/api/endpoints/organization_index.py, line 32 \[r1\]](https://reviewable.io:443/reviews/getsentry/sentry/3924#-KOw1p0V0C8rgRzMxdeG:-KOwCiEPwWTemDSHbMdN:b-7ql1td) ([raw file](https://github.com/getsentry/sentry/blob/3edefbd6fc69e60ee8634a804c504d66fcd12b11/src/sentry/api/endpoints/organization_index.py#L32)):*
<details><summary><i>Previously, mattrobenolt (Matt Robenolt) wrote…</i></summary>
> I know this preexisted, but this field length is only 64 in the databass.
</details>
fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3924)*
<!-- Sent from Reviewable.io -->

mattrobenolt commented:
> I think we should not allow capital letters going forward, but I'm open to discussion. It's a poor experience if you have FOO and foo as unique entities.

Yeah, agreed. So I'm ok with restricting it at the API, we'll just want to make sure then that we adjust this on all forms across the board. But this is fine for now. Was just pointing out the change in behavior.

---

Reviewed 6 of 6 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3924#-:-KOwYRfzRYBYg0d4mEby:b8lioou)*
<!-- Sent from Reviewable.io -->
